Sources/ComposableArchitecture/Reducer.swift

```diff
@@ -290,39 +290,162 @@ public struct Reducer<State, Action, Environment> {
+  ///           Reducer { state, action, environment in
+  ///             ...
+  ///           }
+  ///         )
+  ///
+  ///   * A child effect feeds a child action back into the store when child state is `nil`:
+  ///
+  ///         let childReducer = Reducer<
+  ///           ChildState, ChildAction, ChildEnvironment
+  ///         > { state, action environment in
+  ///           switch action {
+  ///           case .onAppear:
+  ///             // An effect may want to feed its result back to the child domain in an action
+  ///             return environment.apiClient
+  ///               .request()
+  ///               .map(ChildAction.response)
+  ///
+  ///           case let .response(response):
+  ///             // But the child cannot process this action if its state is `nil` in the parent
+  ///           ...
+  ///           }
+  ///         }
+  ///
+  ///     It is perfectly reasonable to ignore the result of an effect when child state is `nil`,
+  ///     for example one-off effects that you don't want to cancel. However, many long-living
+  ///     effects _should_ be explicitly canceled when tearing down a child domain:
+  ///
+  ///         let childReducer = Reducer<
+  ///           ChildState, ChildAction, ChildEnvironment
+  ///         > { state, action environment in
+  ///           struct MotionId: Hashable {}
+  ///
+  ///           switch action {
+  ///           case .onAppear:
+  ///             // Mark long-living effects that shouldn't outlive their domain cancellable
+  ///             return environment.motionClient
+  ///               .start()
+  ///               .map(ChildAction.motion)
+  ///               .cancellable(id: MotionId())
+  ///
+  ///           case .onDisappear:
+  ///             // And explicitly cancel them when the domain is torn down
+  ///             return .cancel(id: MotionId())
+  ///           ...
+  ///           }
+  ///         }
+  ///
+  ///   * A view store sends a child action when child state is `nil`:
+  ///
+  ///         WithViewStore(self.parentStore) { parentViewStore in
+  ///           // If child state is `nil`, it cannot process this action.
+  ///           Button("Child Action") { parentViewStore.send(.child(.action)) }
+  ///           ...
+  ///         }
+  ///
+  ///     Use `Store.scope` with`IfLetStore` or `Store.ifLet` to ensure that views can only send
+  ///     child actions when the child domain is non-`nil`.
+  ///
+  ///         IfLetStore(
+  ///           self.parentStore.scope(state: { $0.child }, action: { .child($0) }
+  ///         ) { childStore in
+  ///           // This destination only appears when child state is non-`nil`
+  ///           WithViewStore(childStore) { childViewStore in
+  ///             // So this action can only be sent when child state is non-`nil`
+  ///             Button("Child Action") { childViewStore.send(.action) }
+  ///           }
+  ///           ...
+  ///         }
```

Comment:
If these additional docs look good I can create similar versions for `forEach`, or I can add a "see also" in `forEach` to read the `optional` documentation for more info (although I think the explanations are slightly different).